code/modules/mob/living/slime/slime_AI.dm

```diff
@@ -0,0 +1,254 @@
+/datum/ai/slime
+	//expected_type = /mob/living/slime
+	var/mood
+	var/chase_target = 0
+	var/mob/living/leader
+	var/mob/living/current_target // Currently attacking this mob (separate from feeding)
+	var/attacked = 0 // Determines if it's been attacked recently. Can be any number, is a cooloff-ish variable
+	var/rabid = 0 // If set to 1, the slime will attack and eat anything it comes in contact with
+	var/list/observed_friends // A list of refs to friends; they are not considered targets for feeding; passed down after splitting.
+	var/list/friendship_cooldown // A list of refs to friends and the next time they can increase friendship.
+	var/list/speech_buffer // Last phrase said near it and person who said it
+	var/mob/living/slime/slime
+	var/next_core_logic_run = 0
+	var/holding_still = 0 // AI variable, cooloff-ish for how long it's going to stay in one place
+
+/datum/ai/slime/New()
+	..()
+	slime = body
+
+/datum/ai/slime/Destroy()
+	observed_friends = null
+	friendship_cooldown = null
+	current_target = null
+	speech_buffer = null
+	slime = null
+	. = ..()
+
+/datum/ai/slime/proc/assess_target(var/mob/living/target)
+	if(!istype(target) || isslime(target) || (weakref(target) in observed_friends))
+		return FALSE
+	if(target.stat != DEAD && (rabid || attacked))
+		return TRUE
+	if(slime.check_valid_feed_target(target) == FEED_RESULT_VALID)
+		return TRUE
+	return FALSE
+
+/datum/ai/slime/proc/update_mood()
+	if(!slime || !body)
+		return
+	body.a_intent_change(I_HELP)
+	var/new_mood
+	if(body.confused)
+		new_mood = "pout"
+	else if(rabid || attacked)
+		new_mood = "angry"
+		body.a_intent_change(I_HURT)
+	else if(current_target)
+		new_mood = "mischevous"
+
+	if(!new_mood)
+		if(prob(1))
+			new_mood = pick("sad", ":3")
+		else if(prob(75))
+			new_mood = mood
+
+	if(new_mood != mood)
+		mood = new_mood
+		body.regenerate_icons()
+
+
+
+
+/*
+	var/newmood = ""
+	a_intent = I_HELP
+	if(confused)
+		newmood = "pout"
+	else if(rabid || attacked)
+		newmood = "angry"
+		a_intent = I_HURT
+	else if(Target)
+		newmood = "mischevous"
+
+	if (!newmood)
+		if (prob(1))
+			newmood = pick("sad", ":3")
+
+	if ((mood == "sad" || mood == ":3") && !newmood)
+		if (prob(75)) newmood = mood
+
+*/
```

Comment:
?

Reply:
Ah copied this in from source for reference when I was fixing some issues with it and forgot to remove it afterward.